File: 20221130/JadenCase/src/main/java/JadenCase.java
public class JadenCase {
    public String solution(String s) {
        String[] words = s.split(" ");

        String answer = "";

        for (int i = 0; i < words.length; i += 1) {
            answer += changeJadenCase(words[i]);

            if (i < words.length - 1) {
                answer += " ";
            }
        }

        if (s.substring(s.length() - 1).equals(" ")) {
            answer += " ";
        }

        return answer;
    }

    public String changeJadenCase(String word) {
        if (word.length() == 0) {
            return "";
        }

        String firstLetter = word.substring(0, 1).toUpperCase();
        String remainLetter = word.substring(1).toLowerCase();

        return firstLetter + remainLetter;
    }
}

File: 20221201/JadenCase.js
export function changeJadenCase(word) {
  const firstWord = word.substring(0, 1).toUpperCase();
  const otherWord = word.substring(1).toLowerCase();

  return firstWord + otherWord;
}

export default function solution(s) {
  return s.split(' ').map((word) => changeJadenCase(word))
    .join(' ');
}
